chain/x/zoracle/client/rest/query.go

```diff
@@ -6,100 +6,118 @@ import (
 	"net/http"
 
 	"github.com/cosmos/cosmos-sdk/client/context"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/rest"
 	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
 	"github.com/gorilla/mux"
 
 	"github.com/bandprotocol/d3n/chain/x/zoracle/internal/types"
 )
 
+// buildTxDetail takes a TxResponse instance and builds new TxDetail contains only necessary fields.
+func buildTxDetail(tx *sdk.TxResponse) TxDetail {
+	return TxDetail{
+		Hash:      tx.TxHash,
+		Height:    tx.Height,
+		Timestamp: tx.Timestamp,
+	}
+}
+
+// buildRequestRESTInfo takes a RequestQuerierInfo instance and builds a more comprehensive version of it.
+func buildRequestRESTInfo(
+	ctx context.CLIContext, id string, queryRequest types.RequestQuerierInfo,
+) (RequestRESTInfo, error) {
+	var request RequestRESTInfo
+
+	request.OracleScriptID = queryRequest.Request.OracleScriptID
+	request.Calldata = queryRequest.Request.Calldata
+	request.RequestedValidators = queryRequest.Request.RequestedValidators
+	request.SufficientValidatorCount = queryRequest.Request.SufficientValidatorCount
+	request.ExpirationHeight = queryRequest.Request.ExpirationHeight
+	request.IsResolved = queryRequest.Request.IsResolved
+	request.RawDataRequests = queryRequest.RawDataRequests
+
+	request.Result = queryRequest.Result
+
+	// Get request detail
+	searchRequest, err := utils.QueryTxsByEvents(
+		ctx,
+		[]string{fmt.Sprintf("%s.%s='%s'", types.EventTypeRequest, types.AttributeKeyRequestID, id)},
+		1,
+		1,
+	)
+	if err != nil || len(searchRequest.Txs) != 1 {
+		return RequestRESTInfo{}, err
+	}
+	request.RequestTx = buildTxDetail(&searchRequest.Txs[0])
+	for _, msg := range searchRequest.Txs[0].Tx.GetMsgs() {
+		if msg.Type() == "request" && msg.Route() == types.RouterKey {
+			request.Requester = msg.(types.MsgRequestData).Sender
+			break
+		}
+	}
+
+	// Save report tx
+	searchReports, err := utils.QueryTxsByEvents(
+		ctx,
+		[]string{fmt.Sprintf("%s.%s='%s'", types.EventTypeReport, types.AttributeKeyRequestID, id)},
+		1,
+		10000, // Estimated validator reports
+	)
+
+	request.Reports = make([]ReportDetail, 0)
+
+	for _, report := range searchReports.Txs {
+		var validatorAddress sdk.ValAddress
+		for _, msg := range report.Tx.GetMsgs() {
+			if msg.Type() == "report" && msg.Route() == types.RouterKey {
```

Comment:
Should we use `MsgReportData{}.Type()` here? 

or alternatively should we just try to cast and check if it's ok? like
```
msgReport, ok := msg.(types.MsgReportData)
if ok {
    ...
}
```

Basically so we don't need to hardcode "report"